xrpl/utils/tx_parser/utils.py

```diff
@@ -0,0 +1,1052 @@
+        Dict[  # Field: 'meta'
+            str,  # 'AffectedNodes'
+            List[
+                Dict[
+                    str,  # Node state
+                    Dict[
+                        str,
+                        Union[str, int, Dict[str, Union[str, int, Dict[str, str]]]],
+                    ],
+                ],
+            ],
+        ],
+    ],
+]
+
+
+# metadata received from a subscribtion method
+SUBSCRIPTION_METADATA_TYPE = Dict[
+    str,
+    Union[
+        str,
+        int,
+        Dict[  # Field: 'transaction'
+            str,
+            Union[
+                str,
+                int,
+                Dict[str, str],  # issued currency amount
+                List[List[Dict[str, Union[str, int]]]],  # Field: 'paths'
+            ],
+        ],
+        Dict[  # Field: 'meta'
+            str,  # 'AffectedNodes'
+            List[
+                Dict[
+                    str,  # Node state
+                    Dict[
+                        str,
+                        Union[str, int, Dict[str, Union[str, int, Dict[str, str]]]],
+                    ],
+                ],
+            ],
+        ],
+    ],
+]
+
+
+class XRPLMetadataException(XRPLException):
+    """Exception for invalid transaction metadata."""
+
+    pass
+
+
+def is_valid_metadata(
+    metadata: Union[METADATA_TYPE, SUBSCRIPTION_METADATA_TYPE],
+) -> None:
+    """Check if the given metadata is valid.
+
+    Args:
+        metadata (Union[METADATA_TYPE, SUBSCRIPTION_METADATA_TYPE]):
+            The transactions metadata
+
+    Raises:
+        XRPLMetadataException: If field Account is missing.
+        XRPLMetadataException: If field meta is missing.
+        XRPLMetadataException: If field AffectedNodes is missing or is empty.
+    """
+    if "transaction" in metadata:
+        if "Account" not in metadata["transaction"]:  # type: ignore
+            raise XRPLMetadataException(
+                "Metadata incomplete: Metadata field 'Account' must be included."
+            )
+    elif "Account" not in metadata:
+        raise XRPLMetadataException(
+            "Metadata incomplete: Metadata field 'Account' must be included."
+        )
+
+    if "meta" not in metadata:
+        raise XRPLMetadataException(
+            "Metadata incomplete: Metadata field 'meta' must be included."
+        )
+
+    meta = metadata["meta"]
+    if "AffectedNodes" not in meta or not meta["AffectedNodes"]:  # type: ignore
+        raise XRPLMetadataException("Metadata incomplete: No nodes provided.")
+
+
+def normalize_metadata(
+    metadata: Union[METADATA_TYPE, SUBSCRIPTION_METADATA_TYPE],
+) -> METADATA_TYPE:
+    """Formats the transaction metadata into one standard format.
+
+    Args:
+        metadata (Union[METADATA_TYPE, SUBSCRIPTION_METADATA_TYPE]):
+            Transactions metadata.
+
+    Returns:
+        Metadata in standard format.
+    """
+    transaction = metadata["transaction"]
+    assert isinstance(transaction, dict)
```

Comment:
You can simplify this logic and avoid the `isinstance` if `METADATA_TYPE` and `SUBSCRIPTION_METADATA_TYPE` are defined with `TypedDict`: https://docs.python.org/3/library/typing.html#typing.TypedDict